Dockerfile

```diff
@@ -1,12 +1,23 @@
 FROM python:3.11.6-slim
-RUN apt update
+
 WORKDIR /usr/src/owaspnettacker
+
 COPY . .
-RUN mkdir -p .data/results
-RUN apt-get update
-RUN apt-get install -y $(cat requirements-apt-get.txt)
-RUN pip3 install --upgrade pip
-RUN pip3 install -r requirements.txt
-RUN pip3 install -r requirements-dev.txt
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN mkdir -p .data/results && \
+    apt-get update && \
+    apt-get install -y $(cat requirements-apt-get.txt) && \
+    pip3 install --upgrade pip && \
+    pip3 install -r requirements.txt && \
+    pip3 install -r requirements-dev.txt && \
+    apt-get clean && \
+    rm -rf /root/.cache/* && \
+    rm -rf /var/lib/apt/lists/*
```

Comment:
I am not sure if that's a good idea, it makes it harder to debug for each step if fails, and more time consuming to repeat the process.